docs_src/async_sql_databases/tutorial001.py

```diff
@@ -58,8 +58,35 @@ async def read_notes():
     return await database.fetch_all(query)
 
 
+@app.get("/notes/{id}/", response_model=Note)
+async def read_one_note(id: int):
+    query = notes.select().where(notes.c.id == id)
+    return await database.fetch_one(query)
+
+
 @app.post("/notes/", response_model=Note)
 async def create_note(note: NoteIn):
-    query = notes.insert().values(text=note.text, completed=note.completed)
-    last_record_id = await database.execute(query)
+    stmt = notes.insert().values(text=note.text, completed=note.completed)
+    last_record_id = await database.execute(stmt)
     return {**note.dict(), "id": last_record_id}
+
+
+@app.put("/notes/{id}/", response_model=Note)
+async def update_note(id: int, note: NoteIn):
+    stmt = (
+        notes.update()
+        .values(text=note.text, completed=note.completed)
+        .where(notes.c.id == id)
+    )
+    await database.execute(stmt)
+    query = notes.select().where(notes.c.id == id)
+    return await database.fetch_one(query)
+
+
+@app.delete("/notes/{id}/")
+async def delete_note(id: int):
+    query = notes.select().where(notes.c.id == id)
+    note = await database.fetch_one(query)
+    stmt = notes.delete().where(notes.c.id == id)
+    await database.execute(stmt)
+    return note
```

Comment:
Would argue about just returning 204 instead of the deleted object, aka
```
@app.delete("/notes/{note_id}/", status_code=204)
async def delete_note(note_id: int):
    query = notes.delete().where(notes.c.id == note_id)
    await database.execute(query)
    return Response(status_code=204)
```
but would like to hear others opinion

Reply:
agreed, there shouldn't be any reason to return a deleted item

Reply:
I usually return 204 for delete endpoints.

Reply:
Yeah I didn't want to use `return Response(status_code=204)` but if i didn't have that I would get this:

```
INFO:     127.0.0.1:64381 - "DELETE /notes/1/ HTTP/1.1" 204 No Content
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 389, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/fastapi/applications.py", line 165, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/applications.py", line 102, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/routing.py", line 550, in __call__
    await route.handle(scope, receive, send)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/routing.py", line 44, in app
    await response(scope, receive, send)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/responses.py", line 128, in __call__
    await send({"type": "http.response.body", "body": self.body})
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/exceptions.py", line 68, in sender
    await send(message)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/starlette/middleware/errors.py", line 156, in _send
    await send(message)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 483, in send
    output = self.conn.send(event)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/h11/_connection.py", line 469, in send
    data_list = self.send_with_data_passthrough(event)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/h11/_connection.py", line 502, in send_with_data_passthrough
    writer(event, data_list.append)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/h11/_writers.py", line 78, in __call__
    self.send_data(event.data, write)
  File "/Users/user/code-utils/test_env/lib/python3.9/site-packages/h11/_writers.py", line 98, in send_data
    raise LocalProtocolError("Too much data for declared Content-Length")
h11._util.LocalProtocolError: Too much data for declared Content-Length
```

happened with both
```
@app.delete("/notes/{note_id}/", status_code=204)
async def delete_note(note_id: int):
    query = notes.delete().where(notes.c.id == note_id)
    await database.execute(query) 
    return None
```
and 
```
@app.delete("/notes/{note_id}/", status_code=204)
async def delete_note(note_id: int):
    query = notes.delete().where(notes.c.id == note_id)
    await database.execute(query) 
```

Reply:
I'm using `httptools` implementation: `uvicorn <app> --http httptools`.

I opened this months ago: https://github.com/encode/uvicorn/issues/841
But I stop looking for the reason and start using `httptools` (is also the implementation that the uvicorn worker for gunicorn uses)... I still think is the reason I stated there...

Reply:
Is there a big difference between h11 and httptools? Find the fact uvicorn uses h11 by default and httptools when in gunicorn worrying. If there is a difference I would maybe suggest changing the docs so we use `-http httptools`?

For now I guess we should keep the `return Response(status_code=204)` so to not add overhead off an extra command on uvicorn run?

Reply:
I didn't stop to look at the implementation itself. I know that h11 is pure python, and httptools is not. But that's my limit here.

I agree with you them, but maybe a note explaining or even going with the 200 here and maybe opening another issue/PR with that info?

Reply:
I prefer going with 204, but don't see an issue of going with 200 here. And yeah we should open issue/PR with the 204 info.

Reply:
Yes, I have removed the code that returns the deleted item and instead have an empty response body.

Also for the simplicity of this example, I have gone with the 200 status code like @Kludex said. However, opening another issue/PR is a good idea and this example can be updated once this bug has been fixed.